src/Handlers/CardsPaymentMethodHandler.php

```diff
@@ -0,0 +1,390 @@
+    /**
+     * @return string
+     */
+    public function getAdyenPaymentMethodId(): string
+    {
+        return 'scheme';
+    }
+
+    /**
+     * @param AsyncPaymentTransactionStruct $transaction
+     * @param RequestDataBag $dataBag
+     * @param SalesChannelContext $salesChannelContext
+     * @param string|null $gateway
+     * @param string $type
+     * @param array $gatewayInfo
+     * @return RedirectResponse
+     */
+    public function pay(
+        AsyncPaymentTransactionStruct $transaction,
+        RequestDataBag $dataBag,
+        SalesChannelContext $salesChannelContext,
+        string $gateway = null,
+        string $type = 'redirect',
+        array $gatewayInfo = []
+    ): RedirectResponse {
+        try {
+            $request = $this->preparePaymentsRequest($salesChannelContext, $transaction);
+        } catch (Exception $exception) {
+            $this->logger->error(
+                sprintf(
+                    "There was an error with the payment method. Order number: %s Missing data: %s",
+                    $transaction->getOrder()->getOrderNumber(),
+                    $exception->getMessage()
+                )
+            );
+            exit;
+        }
+
+        try {
+            $response = $this->checkoutService->payments($request);
+        } catch (AdyenException $exception) {
+            $this->logger->error(
+                sprintf(
+                    "There was an error with the /payments request. Order number %s: %s",
+                    $transaction->getOrder()->getOrderNumber(),
+                    $exception->getMessage()
+                )
+            );
+        }
+
+        return RedirectResponse::create($transaction->getReturnUrl());
+    }
+
+
+    /**
+     * @param AsyncPaymentTransactionStruct $transaction
+     * @param Request $request
+     * @param SalesChannelContext $salesChannelContext
+     */
+    public function finalize(
+        AsyncPaymentTransactionStruct $transaction,
+        Request $request,
+        SalesChannelContext $salesChannelContext
+    ): void {
+        // TODO: Implement finalize() method.
+    }
+
+    //TODO move to util or outsource to lib
+
+    /**
+     * @param string $address
+     * @return array
+     */
+    public function splitStreetAddressHouseNumber(string $address): array
+    {
+        return [
+            'street' => $address,
+            'houseNumber' => 'N\A'
+        ];
+    }
+
+    /**
+     * @param SalesChannelContext $salesChannelContext
+     * @param AsyncPaymentTransactionStruct $transaction
+     * @return array
+     */
+    public function preparePaymentsRequest(
+        SalesChannelContext $salesChannelContext,
+        AsyncPaymentTransactionStruct $transaction
+    ) {
+
+        //Get state.data using the context token
+        $request = json_decode($this->paymentStateDataService->getPaymentStateDataFromContextToken(
+            $salesChannelContext->getToken()
+        )->getStateData(), true);
+
```

Comment:
Let's validate if it is a valid json

```suggestion
    if (json_last_error() !== JSON_ERROR_NONE) {
        // Throw error, log error
    }
```